server/src/test/java/org/opensearch/cluster/routing/allocation/IndexShardHotSpotTests.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing.allocation;
+
+import org.opensearch.cluster.OpenSearchAllocationWithConstraintsTestCase;
+
+public class IndexShardHotSpotTests extends OpenSearchAllocationWithConstraintsTestCase {
+
+    /**
+     * Test single node replacement without active indexing.
+     */
+    public void testNodeReplacement() {
+        setupInitialCluster(5, 1, 5, 1);
+        terminateNodes("node_1");
+        assertForIndexShardHotSpots(false, 4);
+        addNodesWithoutIndexing(1, "new_node_");
+        int movesForModeNone = allocateAndCheckIndexShardHotSpots(false, 5, "new_node_0");
+
+        setupInitialCluster(5, 1, 5, 1);
+        terminateNodes("node_1");
+        assertForIndexShardHotSpots(false, 4);
+        addNodesWithoutIndexing(1, "new_node_");
+        int movesForModeUnassigned = allocateAndCheckIndexShardHotSpots(false, 5, "new_node_0");
+        assertTrue(movesForModeUnassigned <= movesForModeNone);
```

Comment:
Hi @ashwinpankaj, would you mind explaining why you need to do exactly the same operation twice, and compare the `moves`? Thanks!